### PR TITLE
Api implementation/ Start of webapp.

### PR DESCRIPTION
**Pull request by mjrosengrant**
# Run Instructions
```
pip install -r requirements.txt
python manage.py import_symptoms
python manage.py runserver
```

# Thought Process
I decided to use a two app architecture for the project. I created an app `api` for the Symptom and Condition models, and an app called `webapp`. The webapp would be a single page Ajax application, that consumed data provided by the api.

The majority of my time was spent on the `api` app. I thought about using the Django REST framework, but decided it would be overkill. It was much easier to write functional views to return JSON.

I started with the modeling work, and then wrote the import_symptoms management command to load the provided test data into the database. then I worked on the views, and got the API urls routed and rendering. The last piece I got done was incrementing the symptom's search_count value when it's api view was hit.

After I had done all that, I moved onto the webapp. I planned on doing all of the API connections through Ajax. I set up the urls, views, and templates for the a basic one page app. I started digging into the Jquery, and quickly realized this was going to take way too long. I was already past the recommended 4 hour limit. I have left the webapp as a blank page.

If I had more time, I would definitely use it on building out the front end. Since I'm submitting the project as a pull request, I'm taking the educated guess that it will not be run locally. In my mind, it is worth the time setting up a strong API before moving on to the sites that consume it. For the equivalent of half a day's work, I think it is a solid foundation on which the client side application can be built.

# TL;DR
What went well: 2 app architecture, Django setup, API layout and data import.
What was difficult: Ajax requests, front end UI (Due to time constraints)
What I would improve: Spend more time on the front end. Tests are always good too.

### CODE REVIEW

**Comment by heather-buoyhealth** [comment]
Mike,

I'm running into an issue w/ initial setup, running `python manage.py import_symptoms` results in the following:
`  ... mjrosengrant/diagnostic-tool/webapp/urls.py", line 3, in <module>
    import views
ModuleNotFoundError: No module named 'views'
`
Will you update as needed to avoid this error? Thanks!

**Comment by mjrosengrant** [comment]
@heather-buoyhealth I updated the import statement to hopefully fix your problem. I wasn't able to replicate the issue though, so if your problem persists please let me know!

**Comment by heather-buoyhealth** [comment]
@mjrosengrant - the first error has cleared now, but I'm still running into an issue with `python manage.py import_symptoms.` Would you please take a look?
`
mjrosengrant2/diagnostic-tool/api/management/commands/import_symptoms.py", line 16, in handle
    for row in symptom_reader:
_csv.Error: iterator should return strings, not bytes (did you open the file in text mode?)
`

**Comment by mjrosengrant** [comment]
@heather-buoyhealth What version of Python and Django are you using? I am not running into these issues locally, so we're probably using different Python versions. I won't be able to look into this until later tonight, but I will include a virtualenv in the next commit, so that hopefully will get our dev environments in sync.

**Comment by mjrosengrant** [comment]
@heather-buoyhealth I removed the byte flag from the csv reader, so hopefully that solves your problem. 
What I believe is happening is that a long time ago I installed the Anaconda python distribution, which updated a lot of the low level Python settings. Somehow my python shell displays as a CLI for an iPython notebook, so there's definitely some unusual stuff going on with my local setup.  

If you're still unable to run it, I can try wrapping it all in a Docker container, to avoid the "It works on my machine" headache.
